lib/engine/step/special_track.rb

```diff
@@ -76,12 +76,17 @@ def potential_tiles(entity, hex)
           .select { |t| @game.phase.tiles.include?(t.color) && @game.upgrades_to?(hex.tile, t, tile_ability.special) }
       end
 
-      def tile_lay_abilities(entity, &block)
+      def tile_lay_abilities(entity, **kwargs, &block)
         return unless entity&.company?
 
-        ability = @game.abilities(entity, :tile_lay, time: 'sold', &block) if @round.respond_to?(:just_sold_company) &&
-          entity == @round.just_sold_company
-        ability || @game.abilities(entity, :tile_lay, time: 'track', &block)
+        if @round.respond_to?(:just_sold_company) && entity == @round.just_sold_company
+          ability = @game.abilities(entity, :tile_lay, time: 'sold', **kwargs, &block)
+        end
+
+        ability ||
+          @game.abilities(
+            entity, :tile_lay, time: %w[SpecialTrack %current_step% owning_corp_or_turn], **kwargs, &block
```

Comment:
After `ability_right_time?`, this function is probably the next most important logic change in the PR